sim/options.go

```diff
@@ -0,0 +1,184 @@
+package sim
+
+import (
+	"time"
+
+	"github.com/filecoin-project/go-f3/gpbft"
+	"github.com/filecoin-project/go-f3/sim/adversary"
+	"github.com/filecoin-project/go-f3/sim/latency"
+	"github.com/filecoin-project/go-f3/sim/signing"
+)
+
+const (
+	defaultSimNetworkName      = "sim"
+	defaultTipSetGeneratorSeed = 0x264803e715714f95 // Seed from Drand.
+	defaultHonestCount         = 3
+)
+
+var (
+	defaultBaseChain     gpbft.ECChain
+	defaultBeacon        = []byte("beacon")
+	defaultGraniteConfig = gpbft.GraniteConfig{
+		Delta:                2.0,
+		DeltaBackOffExponent: 1.3,
+	}
+	defaultLatencyModel latency.Model
+)
+
+func init() {
+	var err error
+	defaultBaseChain, err = gpbft.NewChain([]byte(("genesis")))
+	if err != nil {
+		panic("failed to instantiate default simulation base chain")
+	}
+	defaultLatencyModel, err = latency.NewLogNormal(time.Now().UnixMilli(), time.Second*5)
+	if err != nil {
+		panic("failed to instantiate default simulation latency model")
+	}
+}
+
+type Option func(*options) error
+
+type options struct {
+	// latencyModel models the cross participant communication latencyModel throughout a
+	// simulation.
+	latencyModel latency.Model
+	// honestCount is the honest participant count. Honest participants have one unit
+	// of power each.
+	honestCount int
+	// Duration of EC epochs.
+	ecEpochDuration time.Duration
+	// Time to wait after EC epoch before starting next instance.
+	ecStabilisationDelay time.Duration
+	// If nil then FakeSigningBackend is used unless overridden by F3_TEST_USE_BLS
+	signingBacked      signing.Backend
+	graniteConfig      *gpbft.GraniteConfig
+	traceLevel         int
+	networkName        gpbft.NetworkName
+	tipSetGenerator    *TipSetGenerator
+	initialInstance    uint64
+	baseChain          *gpbft.ECChain
+	beacon             []byte
+	adversaryGenerator adversary.Generator
+	adversaryCount     uint64
+}
+
+func newOptions(o ...Option) (*options, error) {
+	var opts options
+	for _, apply := range o {
+		if err := apply(&opts); err != nil {
+			return nil, err
+		}
+	}
+	if opts.honestCount == 0 {
+		opts.honestCount = defaultHonestCount
+	}
+	if opts.latencyModel == nil {
+		opts.latencyModel = defaultLatencyModel
+	}
+	if opts.graniteConfig == nil {
+		opts.graniteConfig = &defaultGraniteConfig
+	}
+	if opts.signingBacked == nil {
+		opts.signingBacked = signing.NewFakeBackend()
```

Comment:
BLS envvar override is missing

Reply:
Moved to f3sim command as it is the entrypoint to running a simulation. 

Does that make sense?

Reply:
When working on BLS, I used it to run all the tests with BLS as a sanity check.